itest.py

```diff
@@ -0,0 +1,832 @@
+        system = builder.name.split('.')[1]
+
+        # skip if mapping not provided yet...
+        if system not in tag_name:
+            continue
+
+        # Build up exec args
+        args = [shutil.which('docker'),
+                'build',
+                '--no-cache',
+                '--pull',
+                '-f',
+                str(builder),
+                '-t',
+                f'{IRODS_PLUGIN_BUILDER_NAME}:{tag_name[system]}',
+                str(IRODS_DEV_DIR),]
+
+        # TODO: Use the following line if buildkit ever gets supported
+        # See the following: https://github.com/docker/docker-py/issues/2230
+        # img, logs = client.images.build(path=str(IRODS_DEV_DIR), dockerfile=str(builder), nocache=True, rm=True, labels={IRODS_BUILD_IMAGE_NAME: tag_name[system]})
+
+        # logger.debug(f'img=[{img!r}]')
+        # logger.debug(f'img=[{logs!r}]')
+
+        # Run docker build
+        logger.debug(f"Executing args=[{args!r}]")
+        res = subprocess.run(args, env=docker_env)
+        logger.debug(f'res=[{res!r}] of updating builder for system=[{system!r}]')
+
+def refresh_externals_builders():
+    tag_name = {'ubuntu20': '  ubuntu-20.04',
+                'ubuntu22':   'ubuntu-22.04',
+                'ubuntu24':   'ubuntu-24.04',
+                'debian11':   'debian-11',
+                'debian12':   'debian-12',
+                'rocky9':     'rockylinux-9',
+                'almalinux8': 'almalinux-8',}
+
+    # Get current env
+    docker_env = os.environ.copy()
+
+    # Add buildkit to env
+    docker_env['DOCKER_BUILDKIT'] = '1'
+
+    # Setup environ
+    # client = docker.from_env(environment=docker_env)
+    
+    # For all builders found...
+    for builder in IRODS_DEV_DIR.glob('externals_builder.*'):
+        system = builder.name.split('.')[1]
+
+        # skip if mapping not provided yet...
+        if system not in tag_name:
+            continue
+
+        # Build up exec args
+        args = [shutil.which('docker'),
+                'build',
+                '--no-cache',
+                '--pull',
+                '-f',
+                str(builder),
+                '-t',
+                f'{IRODS_EXTERNALS_BUILDER_NAME}:{tag_name[system]}',
+                str(IRODS_DEV_DIR),]
+
+        # TODO: Use the following line if buildkit ever gets supported
+        # See the following: https://github.com/docker/docker-py/issues/2230
+        # img, logs = client.images.build(path=str(IRODS_DEV_DIR), dockerfile=str(builder), nocache=True, rm=True, labels={IRODS_BUILD_IMAGE_NAME: tag_name[system]})
+
+        # logger.debug(f'img=[{img!r}]')
+        # logger.debug(f'img=[{logs!r}]')
+
+        # Run docker build
+        logger.debug(f"Executing args=[{args!r}]")
+        res = subprocess.run(args, env=docker_env)
+        logger.debug(f'res=[{res!r}] of updating builder for system=[{system!r}]')
+
+# -------
+# SCHEMAS
+# -------
+TEST_MATRIX_SCHEMA = {
```

Comment:
See if we can perform more strict validation of `args` and `os`, `db` combinations.

These are for future issues.